nixos/modules/services/video/wivrn.nix

```diff
@@ -0,0 +1,130 @@
+{ config, pkgs, lib, ... }:
+let
+  inherit (lib) mkAliasOptionModule mkIf mkEnableOption mkPackageOption mkOption mkDefault optionalString getExe' maintainers;
+  cfg = config.services.wivrn;
+  configFormat = pkgs.formats.json { };
+  configFile = configFormat.generate "config.json" cfg.config.json;
+in
+{
+  options = {
+    services.wivrn = {
+      enable = mkEnableOption "WiVRn, an OpenXR streaming application";
+
+      package = mkPackageOption pkgs "wivrn" { };
+
+      openFirewall = mkEnableOption "the default ports in the firewall for the WiVRn server";
+
+      defaultRuntime = mkEnableOption ''
+        WiVRn Monado as the default OpenXR runtime on the system. The config can be found at `/etc/xdg/openxr/1/active_runtime.json`.
+
+        Note that applications can bypass this option by setting an active
+        runtime in a writable XDG_CONFIG_DIRS location like `~/.config`
+      '' // { default = true; };
+
+      highPriority = mkEnableOption "high priority capability for wivrn-server" // { default = true; };
+
+      config = {
+        enable = mkEnableOption "configuration for WiVRn";
+        json = mkOption {
+          type = configFormat.type;
+          description = ''
+            Configuration for WiVRn. The attributes are serialized to JSON in config.json.
+            See https://github.com/Meumeu/WiVRn/blob/master/docs/configuration.md
+          '';
+          default = {
+            scale = 0.8;
+            bitrate = 50000000;
+            encoders = [
+              {
+                encoder = "x264";
+                codec = "h265";
```

Comment:
x264 encoder can not encode into h265, WiVRn will ignore the codec value.

Reply:
What should the default be then? I think using software encoding as a default is fine as it should work for everyone. We can't use Nvenc because its proprietary and Vaapi doesn't cover all GPUs (unless that does include Nvidia?)

Reply:
I may not know NixOS enough, if it is meant as a template for user to modify, you can leave x264 encoder with h264 codec.
If it is meant as a safe default, it is better to not have a configuration file at all and let the application decide at run-time.

Reply:
With it needing to be enabled, I believe it functions more as a template.

Reply:
+0 to no config file by default. Alternatively, you could add a systemd unit that checks the hardware and generates a config file at runtime.

Reply:
That feels a bit out of the scope of this module. No config sounds fine, I will push a change if that's what we agree on

Reply:
No default config is defined so the server can use its own default